src/convert.js

```diff
@@ -0,0 +1,277 @@
+        jSExport.typograhphy = formatTypographyToJS(category.tokens);
+        break;
+      case "spacing":
+        jSExport.spacing = formatSpacingToJS(category.tokens);
+        break;
+      default:
+        console.warn("Unrecognized category type");
+    }
+  });
+  return `module.exports = ` + JSON.stringify(jSExport);
+}
+
+function formatColorsToJS(tokens) {
+  let colors = {};
+  tokens.forEach(token => {
+    colors[token.name] = {
+      hsl: `hsl(${token.value.h}, ${token.value.s}%, ${token.value.l}%)`,
+      h: token.value.h,
+      s: token.value.s,
+      l: token.value.l
+    };
+  });
+  return colors;
+}
+
+function formatTypographyToJS(tokens) {
+  let typograhphy = {};
+  tokens.forEach(token => {
+    typograhphy[token.name] = token.value;
+  });
+  return typograhphy;
+}
+
+function formatSpacingToJS(tokens) {
+  let spacing = {};
+  tokens.forEach(token => {
+    spacing[token.name] = token.value;
+  });
+  return spacing;
+}
+
+function generateTSInterfaces(resolvedTokensByCategory, staticInterfaces) {
+  const interfaceExport = Object.entries(resolvedTokensByCategory).reduce(
+    (resultArray, [key, value]) => {
+      switch (value.category) {
+        case "colors": {
+          resultArray.push(
+            ...generateTSInterfaceCategory(
+              value.tokens,
+              "ColorDesingTokens",
+              "ColorToken"
+            )
+          );
+          return resultArray;
+        }
+        case "typography": {
+          resultArray.push(
+            ...generateTSInterfaceCategory(
+              value.tokens,
+              "TypograhpyDesingTokens",
+              "TypographyToken"
+            )
+          );
+          return resultArray;
+        }
+        case "spacing": {
+          resultArray.push(
+            ...generateTSInterfaceCategory(
+              value.tokens,
+              "SpacingDesingTokens",
+              "ValueUnit"
+            )
+          );
+          return resultArray;
+        }
+        default: {
+          console.warn("Unrecognized category type");
+          return resultArray;
+        }
+      }
+    },
+    []
+  );
+  return staticInterfaces + "\n" + interfaceExport.join("");
+}
+
+function generateTSInterfaceCategory(
+  tokens,
+  categoryInterfaceName,
+  cateogryPropertyInterfaceName
+) {
+  return [
+    `export interface ${categoryInterfaceName} {\n`,
+    ...generateTSInterfaceProperties(tokens, cateogryPropertyInterfaceName),
+    "}\n\n"
+  ];
+}
+
+function generateTSInterfaceProperties(tokens, interfaceName) {
+  let interfaceProperties = [];
+  tokens.forEach(token => {
+    interfaceProperties.push(`  ${token.name}: ${interfaceName};\n`);
+  });
+  return interfaceProperties;
+}
+
+function exportFile(outDir, fileName, data) {
+  const dirname = path.normalize(outDir);
+  try {
+    if (!fs.existsSync(dirname)) {
+      fs.mkdirSync(dirname);
+    }
+    fs.writeFile(`${path.join(dirname, fileName)}`, data, err => {
```

Comment:
Should not need template literals?

Reply:
Fixed, was left there when still using static dir.